packages/api/src/services/notification/notification-builder.ts

```diff
@@ -0,0 +1,118 @@
+import { NotificationType } from '../../models/notification/notification-type';
+import { Notification } from '../../models/notification/notification';
+
+/**
+ * A fluent builder class for constructing {@link Notification} instances.
+ *
+ * The `NotificationBuilder` simplifies the creation of notifications by allowing
+ * a step-by-step configuration of notification attributes such as type, title,
+ * and parameters before generating the final immutable {@link Notification} object.
+ *
+ * @example
+ * ```ts
+ * const notification = NotificationBuilder
+ *   .getNotificationBuilder(NotificationType.INFO, 'USER_CREATED')
+ *   .setTitle('User Created Successfully')
+ *   .setParameters({ username: 'JohnDoe' })
+ *   .getNotification();
+ * ```
+ */
+export class NotificationBuilder {
+  private type: NotificationType = NotificationType.INFO;
+  private code = '';
+  private title = '';
+  private parameters?: Record<string, string>;
```

Comment:
I'm worried about this builder. It basically duplicates the fields of the Notification class which is unexpected/unwanted.